sdks/java/io/solace/src/main/java/org/apache/beam/sdk/io/solace/broker/SempBasicAuthClientExecutor.java

```diff
@@ -0,0 +1,171 @@
+  private final transient HttpRequestFactory requestFactory;
+
+  SempBasicAuthClientExecutor(
+      String host,
+      String username,
+      String password,
+      String vpnName,
+      HttpRequestFactory httpRequestFactory) {
+    this.baseUrl = String.format("%s/SEMP/v2", host);
+    this.username = username;
+    this.messageVpn = vpnName;
+    this.password = password;
+    this.requestFactory = httpRequestFactory;
+  }
+
+  private static String getQueueEndpoint(String messageVpn, String queueName) {
+    return String.format("/monitor/msgVpns/%s/queues/%s", messageVpn, queueName);
+  }
+
+  private static String createQueueEndpoint(String messageVpn) {
+    return String.format("/config/msgVpns/%s/queues", messageVpn);
+  }
+
+  private static String subscriptionEndpoint(String messageVpn, String queueName) {
+    return String.format("/config/msgVpns/%s/queues/%s/subscriptions", messageVpn, queueName);
+  }
+
+  BrokerResponse getQueueResponse(String queueName) throws IOException {
+    String queryUrl = getQueueEndpoint(messageVpn, queueName);
+    HttpResponse response = executeGet(new GenericUrl(baseUrl + queryUrl));
+    return BrokerResponse.fromHttpResponse(response);
+  }
+
+  BrokerResponse createQueueResponse(String queueName) throws IOException {
+    String queryUrl = createQueueEndpoint(messageVpn);
+    ImmutableMap<String, Object> params =
+        ImmutableMap.<String, Object>builder()
+            .put("accessType", "non-exclusive")
+            .put("queueName", queueName)
+            .put("owner", username)
+            .put("permission", "consume")
+            .put("ingressEnabled", true)
+            .put("egressEnabled", true)
+            .build();
+
+    HttpResponse response = executePost(new GenericUrl(baseUrl + queryUrl), params);
+    return BrokerResponse.fromHttpResponse(response);
+  }
+
+  BrokerResponse createSubscriptionResponse(String queueName, String topicName) throws IOException {
+    String queryUrl = subscriptionEndpoint(messageVpn, queueName);
+
+    ImmutableMap<String, Object> params =
+        ImmutableMap.<String, Object>builder()
+            .put("subscriptionTopic", topicName)
+            .put("queueName", queueName)
+            .build();
+    HttpResponse response = executePost(new GenericUrl(baseUrl + queryUrl), params);
+    return BrokerResponse.fromHttpResponse(response);
+  }
+
+  private HttpResponse executeGet(GenericUrl url) throws IOException {
+    HttpRequest request = requestFactory.buildGetRequest(url);
+    return execute(request);
+  }
+
+  private HttpResponse executePost(GenericUrl url, ImmutableMap<String, Object> parameters)
+      throws IOException {
+    HttpContent content = new JsonHttpContent(GsonFactory.getDefaultInstance(), parameters);
+    HttpRequest request = requestFactory.buildPostRequest(url, content);
+    return execute(request);
+  }
+
+  private HttpResponse execute(HttpRequest request) throws IOException {
+    request.setNumberOfRetries(2);
```

Comment:
what consideration here for retry twice? Good to have comment for occurrence of hardcoded numbers

Reply:
I extracted the variable and added a comment.